Xplat/src/main/java/vazkii/botania/common/item/lens/EntropicLens.java

```diff
@@ -8,27 +8,35 @@
  */
 package vazkii.botania.common.item.lens;
 
+import net.minecraft.core.BlockPos;
 import net.minecraft.world.entity.projectile.ThrowableProjectile;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.level.Explosion;
+import net.minecraft.world.phys.BlockHitResult;
 import net.minecraft.world.phys.HitResult;
 
 import vazkii.botania.api.internal.ManaBurst;
+import vazkii.botania.common.block.BotaniaBlocks;
 
 public class EntropicLens extends Lens {
 
 	@Override
 	public boolean collideBurst(ManaBurst burst, HitResult pos, boolean isManaBlock, boolean shouldKill, ItemStack stack) {
 		ThrowableProjectile entity = burst.entity();
 		if (pos.getType() == HitResult.Type.BLOCK) {
+			BlockPos hit = ((BlockHitResult) pos).getBlockPos();
+			if (entity.level.getBlockState(hit).is(BotaniaBlocks.pistonRelay)) {
+				return false;
+			}
+
 			if (!entity.level.isClientSide && !burst.isFake() && !isManaBlock) {
 				entity.level.explode(entity, entity.getX(), entity.getY(), entity.getZ(),
 						burst.getMana() / 50F, Explosion.BlockInteraction.BREAK);
 			}
 			return true;
 		}
 
-		return shouldKill;
+		return false;
```

Comment:
Why did you change this line?

Reply:
`return shouldKill;` makes the warped burst invisible in the client side, 
https://github.com/VazkiiMods/Botania/pull/4210#issuecomment-1304808496

Reply:
i feel like there's a deeper issue here. i suspect the function gets passed a false on the server and a true on the client...

Reply:
you're correct